clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java

```diff
@@ -129,6 +136,22 @@ private static ChannelBuilder create(SecurityProtocol securityProtocol,
                     jaasContexts = new HashMap<>(enabledMechanisms.size());
                     for (String mechanism : enabledMechanisms)
                         jaasContexts.put(mechanism, JaasContext.loadServerContext(listenerName, mechanism, configs));
+
+                    // SSL client authentication is enabled in brokers for SASL_SSL only if listener-prefixed config is specified.
+                    if (listenerName != null && securityProtocol == SecurityProtocol.SASL_SSL) {
+                        String configuredClientAuth = (String) configs.get(BrokerSecurityConfigs.SSL_CLIENT_AUTH_CONFIG);
+                        String listenerClientAuth = (String) config.originalsWithPrefix(listenerName.configPrefix(), true)
+                                .get(BrokerSecurityConfigs.SSL_CLIENT_AUTH_CONFIG);
```

Comment:
Constructing the config for the listener prefix seems a bit wasteful just to check that a config exists. I wonder if this is really necessary and if we could check `listenerName.configPrefix() + BrokerSecurityConfigs.SSL_CLIENT_AUTH_CONFIG` in `config` directly instead. Maybe it does not matter too much here as we do this only once after all. What do you think?

Reply:
We can't do `config.getString(prefix+configName)` because `AbstractConfig` only allows configs from its definition to be obtained using `get` and the prefixed config is not treated as a defined config. I couldn't think of alternative way that doesn't need the originals in some form. As you said, it is only used once when broker starts up, so should be ok. But happy to change if you can think of an alternative.

Reply:
Ah right. I can't think of a better way neither.